share/dkg/pedersen/structs.go

```diff
@@ -161,23 +177,23 @@ type ResponseBundle struct {
 }
 
 // Hash hashes the share index and responses
-func (r *ResponseBundle) Hash() []byte {
+func (b *ResponseBundle) Hash() ([]byte, error) {
 	// first order the response slice in a canonical order
-	sort.SliceStable(r.Responses, func(i, j int) bool {
-		return r.Responses[i].DealerIndex < r.Responses[j].DealerIndex
+	sort.SliceStable(b.Responses, func(i, j int) bool {
+		return b.Responses[i].DealerIndex < b.Responses[j].DealerIndex
 	})
 	h := sha256.New()
-	binary.Write(h, binary.BigEndian, r.ShareIndex)
-	for _, resp := range r.Responses {
-		binary.Write(h, binary.BigEndian, resp.DealerIndex)
-		if resp.Status {
-			binary.Write(h, binary.BigEndian, byte(1))
+	_ = binary.Write(h, binary.BigEndian, b.ShareIndex)
+	for _, resp := range b.Responses {
+		_ = binary.Write(h, binary.BigEndian, resp.DealerIndex)
+		if resp.Status == Success {
+			_ = binary.Write(h, binary.BigEndian, byte(1))
 		} else {
-			binary.Write(h, binary.BigEndian, byte(0))
+			_ = binary.Write(h, binary.BigEndian, byte(0))
 		}
 	}
-	h.Write(r.SessionID)
-	return h.Sum(nil)
+	h.Write(b.SessionID)
```

Comment:
Why not checking the error here?

Reply:
why ignore the errors in the entire function but modyfing its signature to return an error?